### PR TITLE
WIP: Add element highlight to addon interactions

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

## What I did

Second attempt now, basing on `future/base` and using addon-highlight instead of addon-a11y.

Applied the great work from @mikewoudenberg into the codebase. Thanks a lot for doing this!!

image

## How to test

See the demo [here](https://5a375b97f4b14f0020b0cda3-ieownqllxs.chromatic.com/?path=/story/addons-interactions-accountform--verification-password)

- [x] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

## TODO
- [x] Only show the highlight button for valid elements (e.g. hide from `expect(true).toBe(true)` calls)
- [x] Change highlight logic to be independent from addon-a11y
- [ ] Discuss the logic done in the instrumenter (see code comments) 
- [ ] Once selected, the interaction row button should become active (Behaves similar to toolbar buttons)
- [ ] The highlight should behave like a toggle, where you can deactivate the highlighting of a particular element

### CODE REVIEW

File: lib/instrumenter/src/instrumenter.ts [hunk 1]
```diff
@@ -169,6 +171,24 @@ export class Instrumenter {
       else this.initialized = true;
     });
 
+    /**
```
Comment (yannbf) [inline lib/instrumenter/src/instrumenter.ts:174]: Pinging @ghengeveld @shilman for opinions. See the comment below, thank you!

File: code/lib/instrumenter/src/instrumenter.ts [hunk 2]
```diff
@@ -167,6 +169,24 @@ export class Instrumenter {
...
+      const storyState = this.getState(e.storyId);
+      const [element] = Array.from(storyState.callRefsByResult.entries()).find(([_key, value]) => {
+        return value.__callId__ === e.selector;
+      });
+
+      Promise.resolve(element).then((el) => {
+        const selector = getCssSelector(el);
+        this.channel.emit(HIGHLIGHT, { elements: [selector], color: '#6c1d5c' });
+      });
+    });
```
Comment (ghengeveld) [inline code/lib/instrumenter/src/instrumenter.ts:188]: Indeed this does not belong here. Instead it should be in the interactions addon. We can have it use `__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__`. Chromatic does that too.

@JonathanKolnik may have some thoughts about it too.

**Review by ghengeveld — CHANGES_REQUESTED** [review]

File: code/lib/instrumenter/src/instrumenter.ts (hunk 2 above)
Comment (JonathanKolnik) [reply code/lib/instrumenter/src/instrumenter.ts:188]: I agree!

**Comment by ndelangen** [comment]
@ghengeveld would it be possible to transfer this to the apps team's backlog?

**Comment by paulelliott** [comment]
This work was started by @yannbf. Wouldn't he be the best person to finish it up?

**Comment by yannbf** [comment]
> This work was started by @yannbf. Wouldn't he be the best person to finish it up?

What I did was move the great work @mikewoudenberg did into the codebase, updating the necessary code to still be working. This is a great contribution I didn't want to see get lost. I could do the necessary work to finish! However it's not in my current plans.

**Comment by ndelangen** [comment]
The merge conflicts are not resolve-able for me. 😞 

I wonder if this PR is salvageable @yannbf 
It require quite a bit of thought and we don't really have the time-resources to address them.

Perhaps we can ask for a community member to pick up this PR, or retry, based on `next`.

**Comment by valentinpalkovic** [comment]
@yannbf Can you bring the PR in shape if it is still relevant?

**Comment by ndelangen** [comment]
I think we should 
- create a ticket describing the desired behavior
- close this PR
- Ask if someone would be interested in working on the ticket.

This PR can't be worked on by new contributors due to the merge conflicts.
And this feature can't be developed by a new contributor (even though the feature itself isn't all that complicated?), due to this PR being open.
And last but not least @yannbf has no time to bring this to completion right now.

@yannbf shall we close this and create a ticket instead?

**Pull request closed without merging**